tests/console/docker.pm

```diff
@@ -50,47 +50,62 @@ sub run {
     validate_script_output("docker search --no-trunc opensuse", sub { m/This project contains the stable releases of the openSUSE distribution/ });
 
     # images can be pulled from the Docker Hub
-    #   - pull minimalistic alpine image of declared version
+    #   - pull minimalistic alpine image of declared version using tag
     #   - https://store.docker.com/images/alpine
     my $alpine_image_version = '3.5';
     assert_script_run("docker pull alpine:$alpine_image_version", 300);
-    assert_script_run("docker images -q alpine:$alpine_image_version");
-    #   - pull hello-world image, typical docker demo image
+    assert_script_run("docker images alpine:$alpine_image_version | grep alpine");
+    #   - pull typical docker demo image without tag. Should be latest.
     #   - https://store.docker.com/images/hello-world
     assert_script_run("docker pull hello-world", 300);
-    assert_script_run("docker images -q hello-world:latest");
+    assert_script_run("docker images hello-world | grep latest");
 
     # local images can be listed
-    validate_script_output("docker images", sub { m/alpine/ });
+    my $cmd_docker_images = 'docker images';
+    my $listed_images     = script_output($cmd_docker_images);
+    unless ($listed_images =~ m/alpine/ && $listed_images =~ m/hello-world/) {
+        die("error: missing image: $cmd_docker_images");
+    }
+
 
     # containers can be spawned
-    validate_script_output("docker run --name test_1 hello-world",                                                     sub { m/Hello from Docker/ });
-    validate_script_output("docker run --name test_2 alpine:$alpine_image_version /bin/echo Hello world",              sub { m/Hello world/ });
-    validate_script_output("docker ps -a",                                                                             sub { m/test_1/ });
-    validate_script_output("docker ps -a",                                                                             sub { m/test_2/ });
-    validate_script_output("docker run --name test_ephemeral --rm alpine:$alpine_image_version /bin/echo Hello world", sub { m/Hello world/ });
+    assert_script_run('docker run --name test_1 hello-world | grep "Hello from Docker\!"');
+    assert_script_run(qq#docker run --name test_2 alpine:$alpine_image_version /bin/echo Hello world | grep "Hello world"#);
+    assert_script_run(qq#docker run --name test_ephemeral --rm alpine:$alpine_image_version /bin/echo Hello world | grep "Hello world"#);
+    my $cmd_docker_ps_a   = 'docker ps -a';
```

Comment:
`docker ps -a` is the old command. Since you are touching this part of the code, I would advise to use the new way: `docker container ls -a`

Reply:
I didn't know that

Reply:
yeah, the want to 'group' the commands. Type `docker container` and press `TAB` to see the new stuff ;)